docs/diagnostics/PX1033.md

```diff
@@ -0,0 +1,34 @@
+# PX1033
+This document describes the PX1033 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                 | Type  | Code Fix    | 
+| ------ | ------------------------------------------------- | ----- | ----------- | 
+| PX1033 | The DAC does not have an explicit primary key declaration | Warning (ISV Level 3: Informational) | Unavailable | 
+
+## Diagnostic Description
+You can define primary key in DACs. The diagnostic suggests adding primary key definition to a DAC. The warning is shown only for DACs that have either PXCacheName or PXPrimaryGraph attribute.
```

Comment:
There is a couple of other restrictions. Diagnostic will not check abstract DACs and DACs which has only unbound DAC properties (so-called "virtual" DACs, used for custom popups and filters for inquiry screens)

Reply:
We usually add code fix description to the documentation. Here it can be useful to emphasize that code fix will generate not only primary key but also a Find method which will make PK more useful (and it is recommended to declare it)